packages/components/src/templates/next/components/native/Prose/Prose.tsx

```diff
@@ -7,37 +9,42 @@ import OrderedList from "../OrderedList"
 import Table from "../Table"
 import UnorderedList from "../UnorderedList"
 
+const ProseComponent = ({
+  component,
+}: {
+  component: NonNullable<ProseProps["content"]>[number]
+}): JSX.Element => {
+  switch (component.type) {
+    case "divider":
+      return <Divider {...component} />
+    case "heading":
+      return <Heading {...component} />
+    case "orderedList":
+      return <OrderedList {...component} />
+    case "paragraph":
+      return (
+        <BaseParagraph
+          content={getTextAsHtml(component.content)}
+          className="prose-body-base text-base-content"
+        />
+      )
+    case "table":
+      return <Table {...component} />
+    case "unorderedList":
+      return <UnorderedList {...component} />
+  }
```

Comment:
nit: should we add the `never` case here?

Reply:
no need, cause the return type of the component forces a `JSX.Element`. Will fail if the return type is undefined.